src/devices/rosstech_dcu706.c

```diff
@@ -0,0 +1,149 @@
+11 bits/byte: 1 start bit, 0 stop bits and odd parity
+
+*/
+
+static const char transmissionTypeData[] = "Data Transmission";
+static const char transmissionTypeBond[] = "Bond";
+
+
+static uint8_t calculateChecksum(const uint8_t *data, size_t size) {
+    uint8_t checksum = 0;
+
+    for (int bit = 0; bit < 8; bit++) {
+        int count = 0;
+
+        for (size_t i = 0; i < size; i++) {
+            count += (data[i] >> bit) & 1;
+        }
+
+        if (count % 2 == 0) {
+            checksum |= (1 << bit);
+        }
+    }
+
+    return checksum;
+}
+
+
+
+static int rosstech_dcu706_decode(r_device *decoder, bitbuffer_t *bitbuffer)
+{
+
+    // We need 55 bits
+    uint8_t msg[7];
+
+    if (bitbuffer->num_rows != 1
+            || bitbuffer->bits_per_row[0] < 55
+            || bitbuffer->bits_per_row[0] > 300) {
+        decoder_logf(decoder, 2, __func__, "bit_per_row %u out of range", bitbuffer->bits_per_row[0]);
+        return DECODE_ABORT_EARLY; // Unrecognized data
+    }
+
+    uint8_t const preambleDataTransmission[] = {0xDD, 0x40};
+
+    unsigned start_pos = bitbuffer_search(bitbuffer, 0, 0, preambleDataTransmission, 11);
+
+    if (start_pos == bitbuffer->bits_per_row[0]) {
+
+        // The Bond command also contains the temperature
+        uint8_t const preambleBond[] = {0xCD, 0x00};
+
+        start_pos = bitbuffer_search(bitbuffer, 0, 0, preambleBond, 11);
+
+        if (start_pos == bitbuffer->bits_per_row[0]) {
+            return DECODE_ABORT_LENGTH;
+        }
+    }
+
+    if (start_pos + 55 > bitbuffer->bits_per_row[0]) {
+        return DECODE_ABORT_LENGTH;
+    }
+
+    bitbuffer_extract_bytes(bitbuffer, 0, start_pos, msg, sizeof(msg) * 8);
+
+    uint8_t syncType = (msg[0] << 1) | (msg[1] >> 7); //S
+    uint8_t id_high = (msg[1] << 4 | msg[2] >> 4);
+    uint8_t id_low = (msg[2] << 7 | msg[3] >> 1);
+    uint16_t id = (uint16_t)(id_high << 8) | id_low; // I
+    uint8_t temp = msg[4] << 2 | msg[5] >> 6; // T
+    uint8_t checkSum = msg[5] << 5 | msg[6] >> 3; // C
+
+
+    // Create a uint8_t array to hold the extracted values
+    uint8_t extractedData[4];
+    extractedData[0] = syncType;
+    extractedData[1] = id_high;
+    extractedData[2] = id_low;
+    extractedData[3] = temp;
+
+    uint8_t calculatedChecksum = calculateChecksum(extractedData, sizeof(extractedData) / sizeof(extractedData[0]));
+    if (calculatedChecksum != checkSum) {
+        decoder_logf(decoder, 2, __func__, "Sanity Check failed. Expected: %04x, Calculated: %04x. Maybe sanity function calculated wrong!", checkSum, calculatedChecksum);
+        // return DECODE_FAIL_SANITY;
+    }
+
+    uint8_t temp_c = ((temp-32)*5)/9;
+
+    /* clang-format off */
+    data_t *data = data_make(
+        "model",            "Model",              DATA_STRING,   "Rosstech Digital Control Unit DCU-706/Sundance/Jacuzzi",
+        "syncType",         "Transmission Type",  DATA_STRING,   syncType == 0xba ? transmissionTypeData : transmissionTypeBond,    
```

Comment:
should be just `"sync"` with an int of 0/1

Reply:
Removed sync alltogether, since it is always present.